### ISSUE TITLE
[Bug] Serious accessibility issues in RadioButton with a ControlTemplate

### ISSUE THREAD

**Issue by holecekp** [xamarin/Xamarin.Forms#13505]
### Description
RadioButtons without a ControlTemplate works well. Besides mouse, or touch, one can activate them from the keyboard. The user can press Tab key to navigate to the particular RadioButton and the Space key can be used to check the RadioButton. Screenreaders can also handle RadioButtons well as long as they do not have a ControlTemplate. For example, Narrator in Window reads the RadioButton text, the type (that it is a RadioButton), and if it is checked, or not.

However, when you add a ControlTemplate to a RadioButton, it works only using mouse (Windows), or touch (Android). The keyboard cannot be used to navigate to it, or to check it. It is also not presented by screen readers as a RadioButton. Templated RadioButtons are basically impossible to use for those that need a screen reader, and for those that use the keyboard instead of the mouse.

I have tried example from Xamarin.Forms Samples repo (https://github.com/xamarin/xamarin-forms-samples/tree/master/UserInterface/RadioButtonDemos/RadioButtonDemos). I have also tried to fix this by adding IsTabStop="True" and AutomationProperties.IsInAccessibleTree="True"  to various elements (RadioButtons themselves, Labels in them, Frames in the ControlTemplate) but it does not improve the situation much. It has no effect on Windows. On Android, this makes tempated RadioButtons to be focusable by the Tab key, but it is useless since checking them with the Space key still does not work.

As soon as you remove the ControlTemplate, everything works.

### Steps to Reproduce

1. Download RadioButtons test project from Xamarin.Forms Samples repo (https://github.com/xamarin/xamarin-forms-samples/tree/master/UserInterface/RadioButtonDemos/RadioButtonDemos).
2. Run the project on Android and UWP. I have not tested it on iOS.
3. First, select the first page that contains only basic RadioButtons without a ControlTemplate. Try to focus a RadioButton with the Tab key, and check/uncheck it with the Space key. Everything should work. Turn on a screen reader (I have tested Narrator only). Again, everything should work.
4. Now, return to the menu, and select the last page that contains RadioButtons with a ControlTemplate. Try to select one of RadioButtons using only keyboard. It is not possible. Turn on Narrator on Windows and compare the output to the previous page without ControlTemplates. You can see (hear) that templated RadioButtons are not handled correctly by the screen reader.

### Expected Behavior
RadioButtons can be focused and checked/unchecked using keyboard even when a ControlTemplated is used. They should be also accessible to screen readers.

### Actual Behavior
Only RadioButtons without a ControlTemplate can be controlled using keyboard. The RadioButtons with a ControlTemplate can be checked/unchecked only using a mouse, or touch. Keys such as Tab and Space are ignored. Screen readers do not treat RadioButtons with a ControlTemplate as RadioButtons. They do not inform the user neither that it is a RadioButton, nor if it is checked or not. (I have tried only Narrator, but I guess that other screen readers will behave similarly).

### Basic Information

- Version with issue: XF 5.0 SR 1.
- Last known good version: none.
- Platform Target Frameworks: 

### Environment

<details><summary>Show/Hide Visual Studio info</summary>

```
Microsoft Visual Studio Community 2019
Version 16.8.4
VisualStudio.16.Release/16.8.4+30907.101
Microsoft .NET Framework
Version 4.8.03752

Installed Version: Community

Visual C++ 2019   00435-60000-00000-AA088
Microsoft Visual C++ 2019

ASP.NET and Web Tools 2019   16.8.557.25636
ASP.NET and Web Tools 2019

ASP.NET Core Razor Language Services   16.1.0.2052803+84e121f1403378489b842e1797df2f3f5a49ac3c
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2019   16.8.557.25636
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.8.557.25636
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   16.8.557.25636
Azure Functions and Web Jobs Tools

C# Tools   3.8.0-5.20604.10+9ed4b774d20940880de8df1ca8b07508aa01c8cd
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2019 - v2.9.30924.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.113+g422d40002e.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   16.8.43 (00471f8)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.8.1
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.62012.31170
Microsoft SQL Server Data Tools

TypeScript Tools   16.0.21016.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.8.0-5.20604.10+9ed4b774d20940880de8df1ca8b07508aa01c8cd
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   16.8.0-beta.20507.4+da6be68280c89131cdba2045525b80890401defd
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   16.8.000.261 (d16-8@bb99248)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.8.0.507 (remotes/origin/d16-8@e87b24884)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.8.112 (86385a3)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   11.1.0.26 (d16-8/a36ce73)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: 5e9cb6d
    Java.Interop: xamarin/java.interop/d16-8@79d9533
    ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
    SQLite: xamarin/sqlite/3.32.1@1a3276b
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc


Xamarin.iOS and Xamarin.Mac SDK   14.8.0.3 (c51fabee8)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```
</details>

### Reproduction Link
https://github.com/xamarin/xamarin-forms-samples/tree/master/UserInterface/RadioButtonDemos/RadioButtonDemos
or you can use the following zip file because the whole repo is quite large:
[RadioButtonDemos.zip](https://github.com/xamarin/Xamarin.Forms/files/5857705/RadioButtonDemos.zip)

### Workaround

Instead of fancy looking RadioButtons with a custom ControlTemplate, you can use only standard RadioButtons without any template if you are expecting that some of your users could use an assistive technology :-(.
